drivers/sensor/qdec_stm32/qdec_stm32.c

```diff
@@ -138,13 +147,23 @@ static const struct sensor_driver_api qdec_stm32_driver_api = {
 	.channel_get = qdec_stm32_get,
 };
 
+#define QDEC_STM32_ENCODER_MODE_VALUE_CHECK(n)                                                     \
+	COND_CODE_1(DT_INST_NODE_HAS_PROP(n, st_encoder_mode),                                     \
+		    (BUILD_ASSERT(!(DT_INST_PROP(n, st_encoder_mode) & ~TIM_SMCR_SMS),             \
+				  "Encoder mode is not supported by this MCU")),                   \
+		    ());
+
 #define QDEC_STM32_INIT(n)                                                                         \
+	QDEC_STM32_ENCODER_MODE_VALUE_CHECK(n)                                                     \
+                                                                                                   \
 	PINCTRL_DT_INST_DEFINE(n);                                                                 \
 	static const struct qdec_stm32_dev_cfg qdec##n##_stm32_config = {                          \
 		.pin_config = PINCTRL_DT_INST_DEV_CONFIG_GET(n),                                   \
 		.timer_inst = ((TIM_TypeDef *)DT_REG_ADDR(DT_INST_PARENT(n))),                     \
 		.pclken = {.bus = DT_CLOCKS_CELL(DT_INST_PARENT(n), bus),                          \
 			   .enr = DT_CLOCKS_CELL(DT_INST_PARENT(n), bits)},                        \
+		.encoder_mode =                                                                    \
+			DT_INST_PROP_OR(n, st_encoder_mode, QDEC_STM32_ENCODER_MODE_UNSET),        \
```

Comment:
Since you now provide a default value, property will always be there and `DT_INST_PROP_OR` and `QDEC_STM32_ENCODER_MODE_UNSET` related code it no more required

Reply:
Right. Should be fixed now.